web/profiles/joinup/joinup.post_update.php

```diff
@@ -178,3 +180,197 @@ function joinup_post_update_legal() {
+  $configs_to_remove = [
+    'field.storage.node.field_visit_count',
+    'field.storage.rdf_entity.field_download_count',
+    'field.field.rdf_entity.asset_distribution.field_download_count',
+    'field.field.node.discussion.field_visit_count',
+    'field.field.node.document.field_visit_count',
+    'field.field.node.event.field_visit_count',
+    'field.field.node.news.field_visit_count',
+    'joinup_core.matomo_settings',
+  ];
+  array_walk($configs_to_remove, function (string $name) use ($config_factory): void {
+    $config_factory->getEditable($name)->delete();
+  });
+}
+
+/**
+ * Move statistics (part 3): Uninstall stale fields.
+ */
+function joinup_post_update_stats3(): void {
+  $definition_update_manager = \Drupal::entityDefinitionUpdateManager();
+  $download_count_definition = $definition_update_manager->getFieldStorageDefinition('field_download_count', 'rdf_entity');
+  $definition_update_manager->uninstallFieldStorageDefinition($download_count_definition);
+  $visit_count_definition = $definition_update_manager->getFieldStorageDefinition('field_visit_count', 'node');
+  $definition_update_manager->uninstallFieldStorageDefinition($visit_count_definition);
+}
+
+/**
+ * Move statistics (part 4): Remove stale triples.
+ */
+function joinup_post_update_stats4(): void {
+  /** @var \Drupal\Driver\Database\sparql\Connection $sparql_connection */
+  $sparql_connection = \Drupal::service('sparql.endpoint');
+  $sparql_connection->query("WITH <http://joinup.eu/asset_distribution/published>
+DELETE {
+  ?s ?p ?o .
+}
+WHERE {
+  ?s ?p ?o .
+  VALUES ?p { <http://schema.org/userInteractionCount> <http://schema.org/expires> }
+}");
+}
+
+/**
+ * Move statistics (part 5): Search API tasks.
+ */
+function joinup_post_update_stats5(): void {
+  // The 'search_api.index.published' config entity should not be updated by
+  // the configuration management, later in the flow, as changing this config
+  // triggers a reindex but, unfortunately, because of the current config import
+  // tools, it uses the obsolete config data. Instead, we partially disable the
+  // reindex and import here the config entity. Later, in the update flow, we
+  // always trigger a reindex.
+  $config_data = Yaml::decode(file_get_contents(drupal_get_path('module', 'joinup_search') . '/config/install/search_api.index.published.yml'));
+  $state = \Drupal::state();
+  $state->set('search_api.index.published.has_reindexed', TRUE);
+  \Drupal::configFactory()->getEditable('search_api.index.published')->setData($config_data)->save();
+  $state->set('search_api.index.published.has_reindexed', FALSE);
+}
+
+/**
+ * Move statistics (part 6): Update field queue.
+ */
+function joinup_post_update_stats6(array &$sandbox): ?string {
+  $db = \Drupal::database();
+
+  // Make sure that cron is not consuming the queue while we're updating it.
+  $settings = Settings::getAll();
+  $settings['joinup.parse_cached_computed_field'] = FALSE;
```

Comment:
That's true. I was thinking of it. Normally the cron should be stopped during performing updates but I'm not so sure it really happens on PROD. Should I mess with `settings.php`?

Reply:
But, no, we can't. Normally, `settings.php` should not be editable from Drupal.

Reply:
I fixed this by acquiring a falke `cron` lock during running that update. In the case an update is already running we should fail because that is unacceptable. In that case the all crons should be stopped and the deploy should be restarted Da capo